xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -111,7 +73,21 @@ public String getServiceAuthority() {
   @SuppressWarnings("unchecked")
   @Override
   public void start(final Listener2 listener) {
+    BootstrapInfo bootstrapInfo = null;
+    try {
+      bootstrapInfo = bootstrapper.readBootstrap();
+    } catch (Exception e) {
+      listener.onError(Status.UNKNOWN.withDescription("Failed to bootstrap").withCause(e));
```

Comment:
`UNAVAILABLE`? Basically `bootstrapInfo` is not available, and you know the cause `e`.

Reply:
Done.